frontend/src/app/faidare/germplasm-results/germplasm-results.component.ts

```diff
@@ -22,37 +29,35 @@ export type Sort = 'name' | 'accession' | 'species' | 'institute' | 'biological-
   selector: 'dd-germplasm-results',
   templateUrl: './germplasm-results.component.html',
   styleUrl: './germplasm-results.component.scss',
-  imports: [AsyncPipe, TranslateModule, SortableHeaderComponent],
+  imports: [TranslateModule, SortableHeaderComponent],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class GermplasmResultsComponent {
   private exportService = inject(ExportService);
   private downloadService = inject(DownloadService);
   private searchStateService = inject(SearchStateService);
 
-  vm$: Observable<ViewModel>;
-  private downloadingSubject = new BehaviorSubject<null | 'mcpd' | 'plant-material'>(null);
+  private downloading = signal<null | 'mcpd' | 'plant-material'>(null);
+  private model = toSignal(this.searchStateService.getModel());
 
-  constructor() {
-    this.vm$ = combineLatest([this.downloadingSubject, this.searchStateService.getModel()]).pipe(
-      map(([downloading, model]) => ({
-        downloading,
-        sortCriterion: model.searchCriteria.sortCriterion,
-        searchCriteria: model.searchCriteria,
-        documents: model.documents as Page<FaidareDocumentModel> | null
-      }))
-    );
-  }
-
-  faidareUrl(document: FaidareDocumentModel) {
-    return `${document.url}`;
-  }
+  vm: Signal<ViewModel | undefined> = computed(() => {
+    const downloading = this.downloading();
+    const model = this.model();
+    return model
+      ? {
+          downloading,
```

Comment:
OOC: wouldn't it be a tiny bit better to only read `this.downloading()` here? Currently, the computed will re-execute if it changes, even if there is no model right? (maybe it can't really happen here, just as a thought experiment).